### ISSUE TITLE
Eloquent relationship accessors not resolved as magic properties

### ISSUE THREAD

**Issue by alies-dev** [psalm/laravel-psalm-plugin#497]
## Description

When an Eloquent model defines relationships (e.g. `belongsTo`, `hasMany`, `belongsToMany`), Laravel allows accessing the relationship result as a magic property (`$contact->vault` instead of `$contact->vault()->first()`). The plugin does not resolve these relationship accessors as magic properties, causing `UndefinedMagicPropertyFetch` at Psalm errorLevel 1.

## Impact

**284 false positives** across [monicahq/monica](https://github.com/monicahq/monica) (262 project files). This is the largest remaining source of `UndefinedMagicPropertyFetch` after the dev-master plugin's migration-based property resolution (which reduced the count from 1,889 to 321).

Top affected patterns:
- `App\Models\Contact::$vault` (28 occurrences) — `belongsTo` relationship
- `App\Models\User::$account` (26) — `belongsTo` relationship
- `App\Models\Vault::$contacts` (15) — `hasMany` relationship
- `App\Models\Post::$journal` (9) — `belongsTo` relationship
- `App\Models\ContactFeedItem::$contact` (8) — `belongsTo` relationship
- `App\Models\ContactFeedItem::$feedable` (7) — `morphTo` relationship
- `App\Models\ContactImportantDate::$contactImportantDateType` (7) — `belongsTo`

## Reproduction

Found while auditing [monicahq/monica](https://github.com/monicahq/monica) at Psalm errorLevel 1.

**Model definition** (`app/Models/Contact.php:144`):
```php
class Contact extends Model
{
    public function vault(): BelongsTo
    {
        return $this->belongsTo(Vault::class);
    }
}
```

**Usage** (`app/Domains/Contact/ManageContactAddresses/Web/ViewHelpers/ModuleContactAddressesViewHelper.php`):
```php
// This triggers UndefinedMagicPropertyFetch
$addressTypesCollection = $contact->vault->account
    ->addressTypes()
    ->get()
    ->map(fn (AddressType $addressType) => [
        'id' => $addressType->id,
        'name' => $addressType->name,
    ]);
```

Psalm error: `Magic instance property App\Models\Contact::$vault is not defined`

## Expected Behavior

The plugin should recognize that `$contact->vault` returns `Vault|null` (for `BelongsTo`) based on the relationship method's return type. Similarly:
- `belongsTo` / `hasOne` / `morphOne` → `?RelatedModel`
- `hasMany` / `belongsToMany` / `morphMany` → `Collection<int, RelatedModel>`

## Proposed Fix

The plugin could inspect model classes for methods returning `BelongsTo`, `HasMany`, `HasOne`, `BelongsToMany`, `MorphTo`, `MorphMany`, `MorphOne`, etc., and register the corresponding magic properties with appropriate types via a `PropertyExistenceProvider` and `PropertyTypeProvider`.

This would involve:
1. Scanning model methods for relationship return types
2. Registering magic properties matching the method names
3. Inferring the property type from the relationship type + generic parameter
